src/Authentication/Common.Authentication/Factories/AuthenticationFactory.cs

```diff
@@ -48,8 +48,18 @@ public AuthenticationFactory()
 
             TracingAdapter.Information(Resources.AdalAuthConfigurationTrace, configuration.AdDomain, configuration.AdEndpoint, 
                 configuration.ClientId, configuration.ClientRedirectUri, configuration.ResourceClientUri, configuration.ValidateAuthority);
+            IAccessToken token;
```

Comment:
We need to add a change to GetServiceClientCredentials to use one of our new LoginSilentAsync methods if the account type is ServicePrincipal and the CertificateThumbprint property is set